[Release Guide] Document Development Milestones (#722)
submariner-io/submariner-website: src/content/development/contribution-guide/_index.en.md

#### Development Milestones

> The milestone sprints are focused on development, encapsulating work on various [tasks](#tasks) and [bugs](#bugs).

Reviewer: ```suggestion
The milestone sprints are focused on development work for various [tasks](#tasks) and [bugs](#bugs).
```

Project members will work on any planned tasks and bugs, and will also work on unplanned bugs should they arise.

> Any [unplanned work](#unplanned-work) will have to follow the defined guidelines.

Reviewer 2: ```suggestion
Any [unplanned work](#unplanned-work) should follow the defined guidelines.
```

Let's make it either "must" or "should" to be clear.

> Each sprint will end with a milestone release which allows the project community to test newly added features and fixed bugs.

Reviewer 2: ```suggestion
Each sprint will end with a release to allow the community to test new features and fixed bugs.
```

It's a bit of "cognitive load"/"tribal knowledge" to know `rc0` is a milestone release like `m1`/`m2`, but `rc1` not would not be as I understand it (I don't think it's expected there would be a sprint meeting at least, as described below for milestone releases).

The rest just makes the sentence more succinct.

In total, three milestone sprints are planned:

* Two milestones ending with releases of milestone `m1` and milestone `m2`.

> * The last sprint will end with the release of the [release candidate](#release-candidates) `rc0`.

Reviewer 3: ```suggestion
* The last sprint ending with the release of the [release candidate](#release-candidates) `rc0`.
```

> As detailed in the [sprints](#sprints) section, each milestone releases will be followed by a test day and the sprint meetings.

Reviewer 3: ```suggestion
As detailed in the [sprints](#sprints) section, each milestone release will be followed by a test day and the sprint meetings.
```

Do we also want to mention documentation days? Or do we want to avoid “enshrining” them, and reinforce documentation as part of the definition of done instead? (I think there’s value in separate documentation tests since they give a more comprehensive review of the documentation as a whole.)

Reviewer 4: Docs day will be happening after every release just like test day.

Author: whatever makes sense :)

we didnt actually discuss doc days too much so theres no consensus. to me doc days make more sense to do after the rc0, if we do them, since docs should be updated as part of "definition of done" anyhow and doc day  is more of a documentation review and alignment IMHO

also we didnt plan a "doc day" section so if we want to have them, maybe add this info later as it also requires changes in other sections

Reviewer 5: Documentation should be reinforce as part of DoD.
We can discuss documentation for each epic/story at the Sprint review meeting at end of each sprint.
